test/index.js

```diff
@@ -524,6 +535,110 @@ test('equals', function() {
   eq(Z.equals(Just(0).constructor, Maybe), true);
 });
 
+test('lte', function() {
+  eq(Z.lte.length, 2);
+  eq(Z.lte.name, 'lte');
+
+  eq(Z.lte(null, null), true);
+  eq(Z.lte(null, undefined), false);
+  eq(Z.lte(undefined, null), false);
+  eq(Z.lte(undefined, undefined), true);
+  eq(Z.lte(false, false), true);
+  eq(Z.lte(false, true), true);
+  eq(Z.lte(true, false), false);
+  eq(Z.lte(true, true), true);
+  eq(Z.lte(new Boolean(false), new Boolean(false)), true);
+  eq(Z.lte(new Boolean(false), new Boolean(true)), true);
+  eq(Z.lte(new Boolean(true), new Boolean(false)), false);
+  eq(Z.lte(new Boolean(true), new Boolean(true)), true);
+  eq(Z.lte(false, new Boolean(false)), false);
+  eq(Z.lte(new Boolean(false), false), false);
+  eq(Z.lte(true, new Boolean(true)), false);
+  eq(Z.lte(new Boolean(true), true), false);
+  eq(Z.lte(42, 42), true);
+  eq(Z.lte(42, 43), true);
+  eq(Z.lte(43, 42), false);
+  eq(Z.lte(0, 0), true);
+  eq(Z.lte(0, -0), true);
+  eq(Z.lte(-0, 0), true);
+  eq(Z.lte(-0, -0), true);
+  eq(Z.lte(NaN, NaN), true);
+  eq(Z.lte(Infinity, Infinity), true);
+  eq(Z.lte(Infinity, -Infinity), false);
+  eq(Z.lte(-Infinity, Infinity), true);
+  eq(Z.lte(-Infinity, -Infinity), true);
+  eq(Z.lte(NaN, Math.PI), false);
+  eq(Z.lte(Math.PI, NaN), false);
+  eq(Z.lte(new Number(0), new Number(0)), true);
+  eq(Z.lte(new Number(0), new Number(-0)), true);
+  eq(Z.lte(new Number(-0), new Number(0)), true);
+  eq(Z.lte(new Number(-0), new Number(-0)), true);
+  eq(Z.lte(new Number(NaN), new Number(NaN)), true);
+  eq(Z.lte(new Number(Infinity), new Number(Infinity)), true);
+  eq(Z.lte(new Number(Infinity), new Number(-Infinity)), false);
+  eq(Z.lte(new Number(-Infinity), new Number(Infinity)), true);
+  eq(Z.lte(new Number(-Infinity), new Number(-Infinity)), true);
+  eq(Z.lte(new Number(NaN), new Number(Math.PI)), false);
+  eq(Z.lte(new Number(Math.PI), new Number(NaN)), false);
+  eq(Z.lte(42, new Number(42)), false);
+  eq(Z.lte(new Number(42), 42), false);
+  eq(Z.lte(new Date(0), new Date(0)), true);
+  eq(Z.lte(new Date(0), new Date(1)), true);
+  eq(Z.lte(new Date(1), new Date(0)), false);
+  eq(Z.lte(new Date(1), new Date(1)), true);
+  eq(Z.lte(new Date(NaN), new Date(NaN)), true);
+  eq(Z.lte('', ''), true);
+  eq(Z.lte('abc', 'abc'), true);
+  eq(Z.lte('abc', 'xyz'), true);
+  eq(Z.lte(new String(''), new String('')), true);
+  eq(Z.lte(new String('abc'), new String('abc')), true);
+  eq(Z.lte(new String('abc'), new String('xyz')), true);
+  eq(Z.lte('abc', new String('abc')), false);
+  eq(Z.lte(new String('abc'), 'abc'), false);
+  eq(Z.lte([], []), true);
+  eq(Z.lte([1, 2], [1, 2]), true);
+  eq(Z.lte([1, 2, 3], [1, 2]), false);
+  eq(Z.lte([1, 2], [1, 2, 3]), true);
+  eq(Z.lte([1, 2], [2]), true);
+  /* eslint-disable comma-spacing */
+  eq(Z.lte([1,,], [1, 0]), true); // eslint-disable-line no-sparse-arrays
+  /* eslint-enable comma-spacing */
```

Comment:
I think we could use `// eslint-disable-next-line comma-spacing no-sparse-arrays` on the line preceding the `[1,,]` to avoid having comments on three consecutive lines. ;)